### ISSUE TITLE
False positive(?) with no-unsafe-any

### ISSUE THREAD

**Issue by ethanresnick** [palantir/tslint#2981]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
// code snippet
function doesTruthinessCheck(it: any): string {
  if(it) {
    return "john";
  }
  return "jane";
}
```

with `tslint.json` configuration:

```json
"no-unsafe-any": true,
```

#### Actual behavior
Unsafe use of expression of type 'any'.

#### Expected behavior
I'd expect no error, as just checking the truthiness of an `any` typed variable doesn't seem unsafe.

**Comment by ajafff** [comment]
I agree that this is a bug. Checking for truthyness or falsyness should be allowed.
#3008 fixes this
